File: PrimeiraEtapa/Semana4/UIImage/ImagemUIKit01/ViewController.m
//
//  ViewController.m
//  ImagemUIKit01
//
//  Created by bepid on 05/11/14.
//  Copyright (c) 2014 RamiroSeGarante. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)toggleImage:(id)sender {
    //_mImage.hidden = !_mImage.hidden;
    if (_mImage.image == nil) {
        _mImage.image = [UIImage imageNamed:@"ifce.jpg"];
    } else {
        _mImage.image = nil;
    }
}

@end

File: Quarta etapa/Semana 8/Watch Image Communication Exchange/Watch Image Communication Exchange/ViewController.swift
//
//  ViewController.swift
//  Watch Image Communication Exchange
//
//  Created by Antônio Ramon Vasconcelos de Freitas on 07/07/15.
//  Copyright (c) 2015 bepid. All rights reserved.
//

import UIKit

class ViewController: UIViewController , UIImagePickerControllerDelegate, UINavigationControllerDelegate{
    
    @IBOutlet weak var imageView: UIImageView!
    
    let imagePicker = UIImagePickerController()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        imagePicker.delegate = self
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func loadImageButtonTapped(sender: UIButton) {
        imagePicker.allowsEditing = false
        imagePicker.sourceType = .PhotoLibrary
        
        presentViewController(imagePicker, animated: true, completion: nil)
    }
    
    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [NSObject : AnyObject]) {
        if let pickedImage = info[UIImagePickerControllerOriginalImage] as? UIImage {
            imageView.contentMode = .ScaleAspectFit
            imageView.image = pickedImage

            if let group = NSFileManager.defaultManager().containerURLForSecurityApplicationGroupIdentifier("group.br.com.bepid.Watch-Image-Communication-Exchange")?.relativePath {
                let imageData = UIImagePNGRepresentation(pickedImage)
                let myEncodedImageData = NSKeyedArchiver.archivedDataWithRootObject(imageData)
                let path = group.stringByAppendingPathComponent("Library").stringByAppendingPathComponent("Image")
                myEncodedImageData.writeToFile(path, atomically: true)
            }

            dismissViewControllerAnimated(true, completion: nil)
        }
    }
    
    func imagePickerControllerDidCancel(picker: UIImagePickerController) {
        dismissViewControllerAnimated(true, completion: nil)
    }

}

